test/asyncio_tests/test_asyncio_client.py

```diff
@@ -214,6 +220,44 @@ async def test_list_database_names(self):
         self.assertIsInstance(names, list)
         self.assertIn(self.collection.database.name, names)
 
+    @unittest.skipIf(not contextvars, 'this test requires contextvars')
+    @asyncio_test
+    async def test_contextvars_support(self):
+        contextvar = contextvars.ContextVar('variable', default='default')
+
+        class Listener(monitoring.CommandListener):
+            def __init__(self):
+                self.vars = []
+
+            def save_contextvar_value(self):
+                self.vars.append(contextvar.get())
+
+            def started(self, event):
+                self.save_contextvar_value()
+
+            def succeeded(self, event):
+                self.save_contextvar_value()
```

Comment:
Can we change (`set()`) the contextvar value here to 'succeeded' and check it after the await call has returned? Seems more compelling than just saving the same value over and over again.

Reply:
This is actually the use case we want to test. set() doesn't work like that because the executor gets a _copy_ of the context. The listener callbacks can make modifications to the context but those changes won't be reflected in the original context.